### PR TITLE
Support ed25519 SSH key pairs by switching from JSch to Apache MINA SSHd

### PR DESCRIPTION
**Pull request by thachlp**
### Motivation:
Related issue: https://github.com/line/centraldogma/issues/817
Since JGit 5.2, JGit provides an alternative SSH transport implementation based on Apache MINA SSHd, which supports ed25519 which is currently not supported by JSch.
We should make a switch given JSch didn't get any updates since 2018 and ed25519 is now the recommended SSH key pair type.
### Modifications:
- Remove pure sshd lib
- Remove jsch lib
- Add `org.eclipse.jgit.ssh.apache`
- Remove old SSH config
- Update ssh remote URL since it becomes `git@github.com:<user_name>/path`
### Result:
- Using `jgit.ssh.apache` instead of `jsch`
- Fix test

### CODE REVIEW

File: server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/GitMirror.java [hunk 1]
```diff
@@ -343,8 +344,14 @@ private GitWithAuth openGit(File workDir) throws Exception {
...
                 if (username != null) {
-                    jGitUri = "ssh://" + username + '@' + remoteRepoUri().getRawAuthority() +
-                              remoteRepoUri().getRawPath();
+                    if (direction() == LOCAL_TO_REMOTE) {
+                        jGitUri = "ssh://" + username + '@' + remoteRepoUri().getRawAuthority() +
+                                  remoteRepoUri().getRawPath();
+                    } else {
+                        final String[] paths = remoteRepoUri().getRawPath().split("/");
+                        jGitUri = "git@" + remoteRepoUri().getRawAuthority() + ':' + username +
+                                  '/' + paths[paths.length - 1];
```
Comment (thachlp) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/GitMirror.java:353]: ssh pattern is `git@github.com:<user_name>/path`

**Comment by minwoox** [comment]
Hi, @thachlp! Removing the old logic that uses JSch looks good to me. 👍 
Do you have a plan for implementing the logic that uses Apache MINA SSHd?
I'm asking this because we need to prioritize this issue in order to support the ed25519 SSH key for our internal services.
I'm wondering if you have enough time to complete this. If you don't have it, please let us know, so we can take this. 🙇

**Comment by thachlp** [comment]
> Hi, @thachlp! Removing the old logic that uses JSch looks good to me. 👍 Do you have a plan for implementing the logic that uses Apache MINA SSHd? I'm asking this because we need to prioritize this issue in order to support the ed25519 SSH key for our internal services. I'm wondering if you have enough time to complete this. If you don't have it, please let us know, so we can take this. 🙇

Hi @minwoox, feel free to take this, I will follow you to learn more, I am still stuck on understanding how we use public/private keys, with apache-mina I don't see it was used 🙃
I will take another minor issue, thank you

**Comment by minwoox** [comment]
Thanks, @thachlp! Let me take it then. 😆

**Comment by thachlp** [comment]
> Thanks, @thachlp! Let me take it then. 😆

Thanks, too.

**Comment by thachlp** [comment]
We have another pr resolved this

**Pull request closed without merging**